x-pack/plugins/cases/server/connectors/cases/cases_connector_executor.ts

```diff
@@ -752,28 +746,44 @@ export class CasesConnectorExecutor {
     };
   }
 
-  private getCasesTitle(ruleName: string, oracleCounter: number) {
-    const suffix =
-      oracleCounter === INITIAL_ORACLE_RECORD_COUNTER
-        ? '(Auto-created)'
-        : `(${oracleCounter}) (Auto-created)`;
+  private getCasesTitle(
+    params: CasesConnectorRunParams,
+    grouping: GroupedAlerts['grouping'],
+    oracleCounter: number
+  ) {
+    const groupingDescription = Object.entries(grouping)
+      .map(([_, value]) => {
+        return convertValueToString(value);
+      })
+      .join(' & ');
 
-    const ruleNameTrimmed = ruleName.slice(0, MAX_TITLE_LENGTH - suffix.length - 1);
+    const suffix = `${
+      groupingDescription.length > 0 ? ` - Grouping by ${groupingDescription}` : ''
+    }${oracleCounter > INITIAL_ORACLE_RECORD_COUNTER ? ` (${oracleCounter})` : ''} (Auto-created)`;
 
-    return `${ruleNameTrimmed} ${suffix}`;
+    const ruleNameTrimmed = params.rule.name.slice(0, MAX_TITLE_LENGTH - suffix.length - 1);
```

Comment:
super nit: Shall we add `...` at the end of trimmed rule? like `ca stack rule with a very long rule name... - Grouping by host.name (Auto-created)`
Because for long rule name, it is not very clear where it is trimmed.

<details><summary> Case view page </summary>

![Screenshot 2024-04-15 at 11 26 56](https://github.com/elastic/kibana/assets/117571355/53ef353c-1ab1-4593-a284-6e685156c746)
</details>

<details><summary> Case table </summary>

![Screenshot 2024-04-15 at 11 26 34](https://github.com/elastic/kibana/assets/117571355/5719fc9c-0a9a-46da-8aae-413a56d123c4)
</details>

Reply:
Good point! I will do it.

Reply:
Done in [`c23f48f` (#180758)](https://github.com/elastic/kibana/pull/180758/commits/c23f48fbbe23a1067e66081aa70089d0bb6e3fa3)